sim/team-validator.ts

```diff
@@ -926,11 +930,12 @@ export class TeamValidator {
 	) {
 		let eventData: EventInfo | null = null;
 		let eventTemplate = template;
+		const eventLsetData = this.dex.getLearnsetData(eventTemplate.speciesid);
```

Comment:
This is the wrong place for this, `eventTemplate` changes on line 937; that's one of your bugs.